jetty-servlet/src/main/java/org/eclipse/jetty/servlet/BaseHolder.java

```diff
@@ -185,6 +186,32 @@ public synchronized boolean isInstance()
         return _instance != null;
     }
 
+    protected <C, W> C wrap(final C component, final Class<W> wrapperFunctionType, final BiFunction<W, C, C> function)
```

Comment:
I really don't understand why you have this templated over C when the class itself is already templated over T?
When will C and T ever  be different types?

Reply:
Good point.

The `protected <C, W>` portion was the problem with using `<T, W>` as that's a local `T` hides the `BaseHolder<T>` declaration.

I fixed this  in commit c569c87bb561d6caff4923deb45d1c50b938f3e2

Reply:
There was a point in time in this effort where wrap() returned the Wrapped/Wrapper version.
But that doesn't allow identity behavior (no wrap).
Once that was removed there wasn't a need for `<C>` anymore.